iceoryx_posh/test/moduletests/test_popo_chunk_sender.cpp

```diff
@@ -103,32 +108,57 @@ class ChunkSender_test : public Test
+{
+    constexpr uint32_t PAYLOAD_SIZE{SMALL_CHUNK / 2};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{iox::CHUNK_DEFAULT_PAYLOAD_ALIGNMENT};
+    auto maybeChunkHeader = m_chunkSender.tryAllocate(
+        iox::UniquePortId(), PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+    ASSERT_FALSE(maybeChunkHeader.has_error());
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1U));
+}
+
+TEST_F(ChunkSender_test, allocate_OneChunkWithoutCustomHeaderAndLargePayloadAlignmentResultsInLargeChunk)
 {
-    auto maybeChunkHeader = m_chunkSender.tryAllocate(sizeof(DummySample), iox::UniquePortId());
+    constexpr uint32_t PAYLOAD_SIZE{SMALL_CHUNK / 2};
+    constexpr uint32_t PAYLOAD_ALIGNMENT{SMALL_CHUNK};
+    auto maybeChunkHeader = m_chunkSender.tryAllocate(
+        iox::UniquePortId(), PAYLOAD_SIZE, PAYLOAD_ALIGNMENT, CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
     ASSERT_FALSE(maybeChunkHeader.has_error());
-    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1u));
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(1).m_usedChunks, Eq(1U));
+}
+
+TEST_F(ChunkSender_test, allocate_OneChunkWithLargeCustomHeaderResultsInLargeChunk)
+{
+    constexpr uint32_t LARGE_HEADER_SIZE{SMALL_CHUNK};
+    auto maybeChunkHeader = m_chunkSender.tryAllocate(
+        iox::UniquePortId(), sizeof(DummySample), alignof(DummySample), LARGE_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+    ASSERT_FALSE(maybeChunkHeader.has_error());
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(1).m_usedChunks, Eq(1U));
 }
 
 TEST_F(ChunkSender_test, allocate_ChunkHasOriginIdSet)
 {
     iox::UniquePortId uniqueId;
-    auto maybeChunkHeader = m_chunkSender.tryAllocate(sizeof(DummySample), uniqueId);
+    auto maybeChunkHeader = m_chunkSender.tryAllocate(
+        uniqueId, sizeof(DummySample), alignof(DummySample), CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
 
     ASSERT_FALSE(maybeChunkHeader.has_error());
     EXPECT_THAT((*maybeChunkHeader)->originId, Eq(uniqueId));
 }
 
 TEST_F(ChunkSender_test, allocate_MultipleChunks)
 {
-    auto chunk1 = m_chunkSender.tryAllocate(sizeof(DummySample), iox::UniquePortId());
-    auto chunk2 = m_chunkSender.tryAllocate(sizeof(DummySample), iox::UniquePortId());
+    auto chunk1 = m_chunkSender.tryAllocate(
+        iox::UniquePortId(), sizeof(DummySample), alignof(DummySample), CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
+    auto chunk2 = m_chunkSender.tryAllocate(
+        iox::UniquePortId(), sizeof(DummySample), alignof(DummySample), CUSTOM_HEADER_SIZE, CUSTOM_HEADER_ALIGNMENT);
 
     ASSERT_FALSE(chunk1.has_error());
     ASSERT_FALSE(chunk2.has_error());
     // must be different chunks
     EXPECT_THAT(*chunk1, Ne(*chunk2));
-    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(2u));
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(2U));
 }
 
 TEST_F(ChunkSender_test, allocate_Overflow)
```

Comment:
Maybe a test which allocates until overflow and then releases some samples to make space to allocate again would be nice.

Reply:
After some thinking, I came to the conclusion that this is already done by `TEST_F(ChunkSender_test, freeChunk)`. This tests doesn't allocate a new chunk, but ensures that the chunks are freed